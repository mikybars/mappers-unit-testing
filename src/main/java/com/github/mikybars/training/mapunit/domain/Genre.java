package com.github.mikybars.training.mapunit.domain;

public enum Genre {
  MALE, FEMALE
}
