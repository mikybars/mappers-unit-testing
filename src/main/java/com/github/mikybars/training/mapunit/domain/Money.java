package com.github.mikybars.training.mapunit.domain;

public class Money {

  private double amount;

  public double getAmount() {
    return amount;
  }

  public Money amount(double amount) {
    this.amount = amount;
    return this;
  }

  @Override
  public String toString() {
    return "Money{" +
        "amount=" + amount +
        '}';
  }
}
