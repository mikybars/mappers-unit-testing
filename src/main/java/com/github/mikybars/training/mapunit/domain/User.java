package com.github.mikybars.training.mapunit.domain;

public class User {

  private String name;

  private Genre genre;

  private Money balance;

  public String getName() {
    return name;
  }

  public User name(String name) {
    this.name = name;
    return this;
  }

  public Genre getGenre() {
    return genre;
  }

  public User genre(Genre genre) {
    this.genre = genre;
    return this;
  }

  public Money getBalance() {
    return balance;
  }

  public User balance(Money balance) {
    this.balance = balance;
    return this;
  }

  @Override
  public String toString() {
    return "User{" +
        "name='" + name + '\'' +
        ", genre=" + genre +
        ", balance=" + balance +
        '}';
  }
}
