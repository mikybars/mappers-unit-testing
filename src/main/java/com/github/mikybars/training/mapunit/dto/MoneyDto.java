package com.github.mikybars.training.mapunit.dto;

public class MoneyDto {

  private double amount;

  private String currency;

  public double getAmount() {
    return amount;
  }

  public MoneyDto amount(double amount) {
    this.amount = amount;
    return this;
  }

  public String getCurrency() {
    return currency;
  }

  public MoneyDto currency(String currency) {
    this.currency = currency;
    return this;
  }

  @Override
  public String toString() {
    return "MoneyDTO{" +
        "amount=" + amount +
        ", currency='" + currency + '\'' +
        '}';
  }
}
