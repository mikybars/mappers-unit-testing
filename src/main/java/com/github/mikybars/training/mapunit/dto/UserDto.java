package com.github.mikybars.training.mapunit.dto;

public class UserDto {

  private String firstName;

  private String genre;

  private MoneyDto balance;

  public String getFirstName() {
    return firstName;
  }

  public UserDto firstName(String firstName) {
    this.firstName = firstName;
    return this;
  }

  public String getGenre() {
    return genre;
  }

  public UserDto genre(String genre) {
    this.genre = genre;
    return this;
  }

  public MoneyDto getBalance() {
    return balance;
  }

  public UserDto balance(MoneyDto balance) {
    this.balance = balance;
    return this;
  }

  @Override
  public String toString() {
    return "UserDTO{" +
        "firstName='" + firstName + '\'' +
        ", genre='" + genre + '\'' +
        ", balance=" + balance +
        '}';
  }
}
