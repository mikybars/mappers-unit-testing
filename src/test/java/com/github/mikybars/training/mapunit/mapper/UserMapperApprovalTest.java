package com.github.mikybars.training.mapunit.mapper;

import com.github.mikybars.training.mapunit.domain.Genre;
import com.github.mikybars.training.mapunit.domain.Money;
import com.github.mikybars.training.mapunit.domain.User;
import com.github.mikybars.training.mapunit.dto.UserDto;
import org.approvaltests.JsonApprovals;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

class UserMapperApprovalTest {

  UserMapper userMapper;

  @BeforeEach
  void setUp() {
    userMapper = new UserMapperImpl();
  }

  @Test
  void asUserDto() {
    User user = new User()
        .name("Miguel")
        .genre(Genre.MALE)
        .balance(new Money()
            .amount(100)
        );

    UserDto mappedUser = userMapper.asUserDto(user);

    JsonApprovals.verifyAsJson(mappedUser);
  }
}