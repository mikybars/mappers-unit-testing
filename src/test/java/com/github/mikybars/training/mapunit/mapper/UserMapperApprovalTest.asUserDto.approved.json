{
  "firstName": "Miguel",
  "genre": "MALE",
  "balance": {
    "amount": 100.0
  }
}